src/js/components/DataChart/DataChart.js

```diff
@@ -511,39 +540,40 @@ const DataChart = forwardRef(
       return result;
     }, [axis, data, series]);
 
-    const renderValue = (serie, dataIndex, valueArg) => {
-      let value;
-      if (valueArg !== undefined) {
-        if (serie && serie.render) return serie.render(valueArg);
-        value = valueArg;
-      } else {
-        const datum = data[dataIndex];
-        value = datum[serie.property];
-        if (serie && serie.render)
-          return serie.render(value, datum, serie.property);
-      }
-      if (serie) {
-        const dateFormat = dateFormats[serie.property];
-        if (dateFormat) return dateFormat(new Date(value));
-        if (serie.prefix) value = `${serie.prefix}${value}`;
-        if (serie.suffix) value = `${value}${serie.suffix}`;
+    const renderValue = (serie = {}, axisValue, y) => {
```

Comment:
When `direction="horizontal"` and `detail={true}` the detail displays the index instead of the properties value. Need to revisit the logic here.

Reply:
This is handled in https://github.com/grommet/grommet/pull/6856